wgpu-hal/src/dx12/adapter.rs

```diff
@@ -382,6 +382,11 @@ impl super::Adapter {
                 && features1.Int64ShaderOps.as_bool(),
         );
 
+        features.set(
+            wgt::Features::TEXTURE_ATOMIC,
+            shader_model >= naga::back::hlsl::ShaderModel::V5_0,
```

Comment:
SM cannot return less than 5_0 on dx12.

Reply:
got it, adding it to the set of always enabled features in dx at the top